### PR TITLE
feat(Display): Add scrollable display

### PR DESCRIPTION
**Pull request by captainsafia**
I still have to clean up the tests.

### CODE REVIEW

**Comment by rgbkrk** [comment]
Excited for this one.

**Comment by captainsafia** [comment]
OK. This should be good for review.

nteract_expanded

I noticed that we weren't using enzyme in the tests for Display. I assume this is a consequence of bringing the DisplayArea in. Do we want to move the tests over to using enzyme so it is consistent with the rest of the code base?

**Comment by rgbkrk** [comment]
> I noticed that we weren't using enzyme in the tests for Display. I assume this is a consequence of bringing the DisplayArea in. Do we want to move the tests over to using enzyme so it is consistent with the rest of the code base?

Yeah, that's a consequence of bringing DisplayArea in. We should definitely switch to `enzyme`.

File: src/notebook/actions.js [hunk 1]
```diff
@@ -251,3 +251,10 @@ export function setConfigKey(key, value) {
 export function setTheme(theme) {
   return setConfigKey('theme', theme);
 }
+
+export function toggleOutputExpansion(id) {
+  return {
+    type: constants.TOGGLE_OUTPUT_EXPANSION,
+    id,
+  };
```
Comment (rgbkrk) [inline src/notebook/actions.js:259]: This one isn't covered by the tests and should be a quick and easy action creator test.

File: src/notebook/components/cell/toolbar.js [hunk 2]
```diff
@@ -87,17 +89,21 @@ export default class Toolbar extends React.Component {
     this.context.store.dispatch(changeCellType(this.props.id, to));
   }
 
+  toggleOutputExpansion() {
+    this.context.store.dispatch(toggleOutputExpansion(this.props.id));
```
Comment (rgbkrk) [inline src/notebook/components/cell/toolbar.js:93]: Use `enzyme`'s instance to get the React Element then run this function.

**Review by rgbkrk — CHANGES_REQUESTED** [review]

File: src/notebook/reducers/document.js [hunk 3]
```diff
@@ -240,4 +241,10 @@ export default handleActions({
       .delete(['notebook', 'cellMap', id, 'execution_count'])
       .delete(['notebook', 'cellMap', id, 'outputs']);
   },
+  [constants.TOGGLE_OUTPUT_EXPANSION]: function toggleOutputExpansion(state, action) {
+    const { id } = action;
+
+    return state.updateIn(['notebook', 'cellMap'], (cells) => cells.setIn([id, 'outputExpanded'],
+      !cells.getIn([id, 'outputExpanded'])));
```
Comment (rgbkrk) [inline src/notebook/reducers/document.js:248]: This looks pretty testable as a reducer.

**Comment by captainsafia** [comment]
> Yeah, that's a consequence of bringing DisplayArea in. We should definitely switch to enzyme.

Yep. I can address that and the coverage drop in another PR.

File: src/notebook/components/cell/display-area/display.js [hunk 4]
```diff
@@ -12,29 +12,41 @@ type Props = {
...
-            theme={props.theme}
-          />
-        )
-      }
-    </div>
-  );
+  const style = {
+    height: props.expanded ? '300px' : 'auto',
+    overflow: props.expanded ? 'scroll' : 'overflow',
+  };
```
Comment (rgbkrk) [inline src/notebook/components/cell/display-area/display.js:25]: This is pretty awesome. I wonder if we could do something where it doesn't scroll unless you focus on it too.

**Pull request merged**